src/time.jl

```diff
@@ -117,6 +117,13 @@ function fpswhen(switch, rate; name=auto_name!("$rate fpswhen", switch))
     n
 end
 
+function fpswhen(switch, rate::Signal; name=auto_name!("fpswhen_rate", rate, switch))
+    #Dispatch for the case where the rate is a time-varying signal.
+    n = Signal(Float64, 0.0, (switch,rate ,); name=name)
```

Comment:
Having `rate` as a parent of `n` seems reasonable. I'm curious, was it needed, or you just thought it was sensible?

Reply:
The latter. I haven't tried it without rate as a parent though.